.yarn/sdks/typescript/lib/tsserver.js

```diff
@@ -91,9 +91,25 @@ const moduleWrapper = tsserver => {
   }
 
   function fromEditorPath(str) {
-    return process.platform === `win32`
-      ? str.replace(/^\^?zip:\//, ``)
-      : str.replace(/^\^?zip:/, ``);
+    switch (hostInfo) {
+      case `coc-nvim`:
+        str = str.replace(/\.zip::/, `.zip/`);
+        // The path for coc-nvim is in format of /<pwd>/zipfile:/<pwd>/.yarn/...
+        // So in order to convert it back, we use .* to match all the thing
+        // before `zipfile:`
+        return process.platform === `win32`
+          ? str.replace(/^.*zipfile:\//, ``)
+          : str.replace(/^.*zipfile:/, ``);
+      case `neovim`:
+        str = str.replace(/\.zip::/, `.zip/`);
+        return process.platform === `win32`
+          ? str.replace(/^\^?zipfile:\//, ``)
```

Comment:
As per below [comment](https://github.com/yarnpkg/berry/pull/3442/files#r712023067) I think the handling of prepended `^` is a vscode requirement so I _think_ it could be omitted here